modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2TreeIndex.java

```diff
@@ -1052,4 +1085,13 @@ public H2Tree create(
             PageIoResolver pageIoRslvr
         ) throws IgniteCheckedException;
     }
+
+    /**
+     * Returns a function that is used to re-create this index.
+     *
+     * @return Function that re-creates index.
+     */
+    public ThrowableBiFunction<PageMemory, IgniteCacheOffheapManager, H2TreeIndex, IgniteCheckedException> getRecreator() {
```

Comment:
Let's remake it into a regular method, it seems we can get everything from the index and its segments.